x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_events_query.ts

```diff
@@ -53,30 +73,34 @@ const buildTimeRangeFilter = ({
                 },
               },
             },
-            {
-              bool: {
-                filter: [
-                  {
-                    range: {
-                      '@timestamp': {
-                        lte: to,
-                        gte: from,
-                        format: 'strict_date_optional_time',
-                      },
-                    },
-                  },
+            ...(shouldDisableTimestampFallback
```

Comment:
If we write this function in terms of `primaryTimestamp` and `secondaryTimestamp`, we can avoid duplicating this logic that handles `shouldDisableTimestampFallback` and the function would look like (truncated)
```
return secondaryTimestamp != null
    ? {
        bool: {
          minimum_should_match: 1,
          should: [
            {
              range: {
                [primaryTimestamp]: {
...
```

Reply:
done